### PR TITLE
test: adding code coverage

### LINKED ISSUE #27: Add test coverage
It would be nice to know how much test coverage we have today.

### PR DESCRIPTION
**Pull request by tedconn**
Fixes #27 

Coverage is generated into `/coverage` (ignored) and currently looks like this:

```
----------|---------|----------|---------|---------|--------------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s        
----------|---------|----------|---------|---------|--------------------------
All files |     100 |     91.3 |     100 |     100 |                          
 index.js |     100 |     91.3 |     100 |     100 | 15,43,78,126,156,234-242 
----------|---------|----------|---------|---------|--------------------------
```

I don't understand how the line coverage can be 100% but the branch 91.3%. I removed some tests to see how the report changes and it was exactly what you would expect, so I am confident this is accurate.

In addition, I added a threshold check which will cause the test runner to fail if the threshold is not met.

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -16,13 +16,30 @@ module.exports = function (config) {
...
     preprocessors: {
       'test/**/*.spec.js': ['rollup']
     },
 
+    coverageReporter: {
+      type: 'text',
+      dir: 'coverage/',
+      file: 'coverage.txt',
+      check: {
+        global: {
```
Comment (tedconn) [inline karma.conf.js:34]: I could reduce these numbers to like 95?

Comment (nolanlawson) [reply karma.conf.js:34]: @tedconn I'd prefer to have everything at 100 and then use `/* istanbul ignore */` comments in the code for stuff we don't feel we need to cover (e.g. IE-only code).

**Comment by nolanlawson** [comment]
@tedconn See https://github.com/tedconn/kagekiri/pull/1 , you can use this to add an HTML reporter, which is useful for debugging. When I open it up, I can see some branches that aren't coverage (but maybe should be – or else ignored).

File: karma.conf.js (hunk 1 above)
Comment (nolanlawson) [reply karma.conf.js:37]: In the short term though, yeah let's just keep it at 91 or something until we can get everything to 100.

**Review by nolanlawson — CHANGES_REQUESTED** [review]
See the PR I opened

**Review by nolanlawson — APPROVED** [review]

**Comment by nolanlawson** [comment]
This is awesome! Thanks @tedconn

**Pull request merged**